合约查询无结果响应

环境：开发
错误：发送合约查询请求成功，但没有收到查询结果回调
分析：Trader与服务端断连后重新连接，但未触发用户重登录
错误日志：

```
[ERROR] 2020-08-12 08:57:46.705 [Thread-590] CTPTraderSpi - [onFrontDisconnected] 客户端与交易服务断开连接. 原因:接收心跳超时.
[INFO ] 2020-08-12 08:59:20.879 [Thread-591] CTPTraderSpi - [onFrontConnected] 连接交易前置机成功!
[DEBUG] 2020-08-12 08:59:31.511 [queryInstrumentListner-0-C-1] CTPRequestManager - [registerUUID] 注册请求UUID. idOrRef:5; uuid:fe440ece-28d5-4ad8-b676-40c4bc4c3978
[DEBUG] 2020-08-12 08:59:31.511 [queryInstrumentListner-0-C-1] CTPRequestManager - [addListener] 添加回调监听. requestID:5
[INFO ] 2020-08-12 08:59:31.512 [queryInstrumentListner-0-C-1] CTPTraderApi - [reqQryInstrument] 发送查询合约请求成功
[DEBUG] 2020-08-12 09:05:11.556 [queryInstrumentListner-0-C-1] CTPRequestManager - [registerUUID] 注册请求UUID. idOrRef:6; uuid:230bde5d-7e64-4c34-b848-3e6c4381e3b8
[DEBUG] 2020-08-12 09:05:11.557 [queryInstrumentListner-0-C-1] CTPRequestManager - [addListener] 添加回调监听. requestID:6
[INFO ] 2020-08-12 09:05:11.557 [queryInstrumentListner-0-C-1] CTPTraderApi - [reqQryInstrument] 发送查询合约请求成功
```

具体代码：
[CTPTrader.java](https://github.com/JupiterFund/trader-ctp/blob/master/src/main/java/com/nodeunify/jupiter/trader/ctp/CTPTrader.java#L72)
[CTPTraderSpi.java](https://github.com/JupiterFund/trader-ctp/blob/master/src/main/java/com/nodeunify/jupiter/trader/ctp/impl/CTPTraderSpi.java#L55)

推荐调试方法：

1. 本地成功运行Trader-CTP
2. 等用户成功登录，并完成一次合约查询成功后，强制断开本地网络连接。模拟断连的错误。
3. 在上述具体代码为止设置调试点
4. 确认用户重新登录是否被触发

--- a/src/main/java/com/nodeunify/jupiter/trader/ctp/CTPTrader.java
+++ b/src/main/java/com/nodeunify/jupiter/trader/ctp/CTPTrader.java
@@ -70,8 +70,7 @@ public class CTPTrader {
     @EventListener
     void handleCTPTraderEvent(CTPTraderEvent event) {
         if ("onFrontReconnected".equals(event.getSource())) {
-            log.debug("[handleCTPTraderEvent] 交易前置已重连，用户重新登录");
-            // bug #1
+            log.debug("[handleCTPTraderEvent] 交易前置已重连，用户重新认证登录");
             authenticate().thenCompose(nil -> login());
         }
         

--- a/src/main/java/com/nodeunify/jupiter/trader/ctp/CTPTrader.java
+++ b/src/main/java/com/nodeunify/jupiter/trader/ctp/CTPTrader.java
@@ -71,7 +71,8 @@ public class CTPTrader {
     void handleCTPTraderEvent(CTPTraderEvent event) {
         if ("onFrontReconnected".equals(event.getSource())) {
             log.debug("[handleCTPTraderEvent] 交易前置已重连，用户重新登录");
-            login();
+            // bug #1
+            authenticate().thenCompose(nil -> login());
         }
         
     }

